components/src/components/top-button/top-button.tsx

```diff
@@ -0,0 +1,31 @@
+import { Component, Listen, Prop, h } from '@stencil/core';
+
+// Scroll to top button.
+@Component({
+  tag: 'pulumi-top-button',
+  styleUrl: 'top-button.scss',
+  shadow: false
+})
+
+export class TopButton {
+
+  @Prop({ mutable: true })
```

Comment:
So this is a Stencil thing, but I think what you want here is [`State`](https://stenciljs.com/docs/state) rather than `Prop`, since `scroll-position` isn't exposed as an API on this component, and in any case is not settable.